lib/automatic/api_client/client_delegator.rb

```diff
@@ -0,0 +1,21 @@
+module Automatic
+  class ApiClient
+    class ClientDelegator
+      attr_reader :api_client, :delegated_class
+
+      def initialize(api_client, delegated_class)
+        @api_client = api_client
+        @delegated_class = delegated_class
+      end
+
+      def method_missing(method_sym, *arguments, &block)
+        super unless delegated_class.respond_to?(method_sym)
+
+        arguments[1 + delegated_class.method(method_sym).arity] ||= {}
```

Comment:
Yeah a lot of this was driven by the desire to maintain as much backwards compatibility as possible. Will see what I can do.